api/src/test/java/org/openmrs/module/mksreports/reports/OutpatientConsultationReportManagerTest.java

```diff
@@ -241,61 +241,56 @@ public void evaluate_shouldProduceDataset() throws Exception {
 			Map<String, Object> obsSummaryMap = row.getColumnValuesByKey();
 			Set<String> colNames = new HashSet<>(obsSummaryMap.keySet());
 			
-			colNames.remove(OutpatientConsultationReportManager.col12);
+			colNames.remove(OutpatientConsultationReportManager.col12); // _25To50, females
 			{
-				List<Integer> _25To50yFemalesForAllDiagnosis = (List<Integer>) row
-				        .getColumnValue(OutpatientConsultationReportManager.col12);
-				assertThat(_25To50yFemalesForAllDiagnosis, is(notNullValue()));
-				assertThat(_25To50yFemalesForAllDiagnosis.size(), is(2));
-				assertThat(_25To50yFemalesForAllDiagnosis.contains(77), is(true));
-				assertThat(_25To50yFemalesForAllDiagnosis.contains(8), is(true));
+				List<Integer> cohort = (List<Integer>) row.getColumnValue(OutpatientConsultationReportManager.col12);
+				assertThat(cohort, is(notNullValue()));
+				assertThat(cohort.size(), is(2));
+				assertThat(cohort.contains(77), is(true));
+				assertThat(cohort.contains(8), is(true));
 			}
 			
-			colNames.remove(OutpatientConsultationReportManager.col7);
+			colNames.remove(OutpatientConsultationReportManager.col7); // _5To15y, males
 			{
-				List<Integer> _5To15yMalesForAllDiagnosis = (List<Integer>) row
-				        .getColumnValue(OutpatientConsultationReportManager.col7);
-				assertThat(_5To15yMalesForAllDiagnosis, is(notNullValue()));
-				assertThat(_5To15yMalesForAllDiagnosis.size(), is(5));
+				List<Integer> cohort = (List<Integer>) row.getColumnValue(OutpatientConsultationReportManager.col7);
+				assertThat(cohort, is(notNullValue()));
+				assertThat(cohort.size(), is(5));
 				
-				Set<Integer> _5To15yMalesForAllDiagnosisWithoutDuplicates = new HashSet<>(_5To15yMalesForAllDiagnosis);
-				assertThat(_5To15yMalesForAllDiagnosisWithoutDuplicates.size(), is(2));
-				assertThat(_5To15yMalesForAllDiagnosisWithoutDuplicates.contains(2), is(true));
-				assertThat(_5To15yMalesForAllDiagnosisWithoutDuplicates.contains(6), is(true));
+				Set<Integer> cohortNoDuplicates = new HashSet<>(cohort);
+				assertThat(cohortNoDuplicates.size(), is(2));
+				assertThat(cohortNoDuplicates.contains(2), is(true));
+				assertThat(cohortNoDuplicates.contains(6), is(true));
```

Comment:
@ivange94 can you explain what is the logic behind those asserts with `cohortNoDuplicates`?

Reply:
I was just trying to show that all edge cases where covered. For example, I had tests where the size of list is 5 but I was only asserting for the presence of 2 person IDs. This is because though the size of the list was 5, it had only 2 unique patients. All test cases where size of list was n, I had to assert the presence of n unique person IDs. But here I'm just showing that, they're not always unique.